### PR TITLE
Added get default_branch

### LINKED ISSUE #1: Add support for branches different than master
Currently, we assume that all the repos have the master as the main branch.

### PR DESCRIPTION
**Pull request by renatosuero**
closes #1 

1 important thing to check, I've moved github and gitlab to source. I did it to "encapsulate all things regarding the source into the source package", I've created generic struct for all source use. Now we return a struct instead of []string

### CODE REVIEW

**Comment by renatosuero** [comment]
@filhodanuvem github actions we check folder by folder, why not run go test -v ./... instead add folders ?

File: stats/stats.go [hunk 1]
```diff
@@ -87,7 +88,7 @@ func getStatsFromRepo(repo, tempPath, provider string, l *log.Logger) (repositor
 
 	tDownloader := downloader.(repository.Downloader)
 
-	path, err = tDownloader.Download(repo, tempPath, l)
+	path, err = tDownloader.Download(repo.URL, tempPath, repo.DefaultBranch, l)
```
Comment (filhodanuvem) [inline stats/stats.go:91]: why not pass repo an access `repo.URL` and `repo.DefaultBranch` ?

**Review by filhodanuvem — COMMENTED** [review]
Thank you for the effort @renatosuero ! 
Answering your question, at the beginning I tried to skip some folders otherwise the `-v ./...` didn't work. If you notice that it's ok to use it now, it sounds good for me to change.

File: stats/stats.go (hunk 1 above)
Comment (renatosuero) [reply stats/stats.go]: totally agree with that :)

File: source/source.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package source
+
+type ProviderRepos struct {
```
[the remark this answers is not in the document]
Comment (renatosuero) [reply source/source.go]: yeah make sense

File: source/gitlab/repositories.go [hunk 3]
```diff
@@ -28,7 +31,10 @@ func GetRepositories(username string) ([]string, error) {
 	json.Unmarshal(body, &jsonResponse)
 
 	for _, repo := range jsonResponse {
-		repos = append(repos, repo.URL)
+		repos = append(repos, source.ProviderRepos{
+			URL:           repo.URL,
+			DefaultBranch: repo.DefaultBranch,
```
[the remark this answers is not in the document]
Comment (renatosuero) [reply source/gitlab/repositories.go:36]: 😍

**Review by filhodanuvem — APPROVED** [review]

**Comment by filhodanuvem** [comment]
Thank you @renatosuero 🎉

**Pull request merged**